core/templates/domain/exploration/read-only-exploration-backend-api.service.ts

```diff
@@ -76,12 +82,28 @@ export class ReadOnlyExplorationBackendApiService {
       const explorationDataUrl = this._getExplorationUrl(
         explorationId, version);
 
-      this.http.get<FetchExplorationBackendResponse>(
-        explorationDataUrl).toPromise().then(response => {
-        resolve(response);
-      }, errorResponse => {
-        reject(errorResponse.error.error);
-      });
+      // If version is not null, then check whether the data is already cached.
```

Comment:
In this comment, what is "the data" referring to? Please be as specific as possible in comments.